draft-west-cookie-incrementalism.md

```diff
@@ -224,6 +233,48 @@ attribute (Section 4.1.2.5 of {{RFC6265bis}}) by altering the storage model defi
 This is conceptually similar to the requirements put into place for the `__Secure-` prefix (Section
 4.1.3.1 of {{RFC6265bis}}).
 
+## Schemeful Same-Site {#schemeful-samesite}
+
+By using the scheme, as well as the registrable domain, schemeful same-site can
+help to protect https origins against a network attacker that is impersonating
+an http origin with the same registrable domain, thereby further increasing
+SameSite's CSRF protections. To do so we need to modify a number of things:
+
+First change the definition of "site for cookies" from a registrable domain to
+an origin. In the places where a we return an empty string for a non-existent
+"site for cookies" we should instead return an origin set to a freshly
+generated globally unique identifier.
+
+Then replace the same-site calculation algorithm with the following:
+
+~~~
+Two origins, A and B, are considered same-site if the following algorithm returns true:
+1.  If A and B are both scheme/host/port triples then
+
+    1.  If A's scheme does not equal B's scheme, return false.
+
+    2.  Let hostA be A's host, and hostB be B's host.
+
+        1.  If hostA equals hostB and hostA's registrable domain is null, return true.
```

Comment:
1. These shouldn't be nested, as 2 isn't a conditional or loop.

2. I think this aims to deal with `github.io` or `appspot.com` hosting content, and being same-site with themselves? What should we do when `yandexcloud.net` makes a request to `storage.yandexcloud.net` (both of which are on the PSL)? What about `cust.dev.thingdust.io` (on the PSL) making a request to `thingdust.io` (not on the PSL)?

Reply:
1. Done.
2. Yeah, this part is to allow for same-siteness of the same PSL entities.
To your other points I have no clue what to do beyond a gut reaction of "If they went out of their way to get different domains added to the PSL they should have to deal with SameSite=None". Maybe @davidben would have some more informed opinions on this? 
If we do decide on doing something different than what's already written there then we should let the HTML folks know because that algorithm is lifted from the HTML spec.

Reply:
I think this definition and HTML's should match. Ideally we'd simply cite HTML, but I do not know what the status of IETF vs WHATWG silliness is. Although having to redefine all WHATWG definitions would make the IETF an hindrance to interop, which would be a spectacular failure of the spec venue to do its core job. So maybe it is worth cutting through the red tape and getting that citation to work.

Regarding prefixes of PSL entries, I think same-site should be closed under transitivity, otherwise the concept can't be used in places like storage isolation. Thus we need to say that `https://example` and `https://etld.example` are cross-site from `https://foo.etld.example` given that we want `https://foo.etld.example` and `https://bar.etld.example` to be cross-site.